### PR TITLE
Adds Vote for round 

### PR DESCRIPTION
**Pull request by mambisi**
feature: Add vote round extrinsic
* Adds ` vote(origin, round_id: T::Hash, amount: T::Balance, vote_multiplier: u8, approve : bool)` to PolkadexIDO
* Exposes `polkadexido_getVotingStat` RPC call to get vote stats

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -1150,6 +1150,7 @@ parameter_types! {
     pub const GetIDOPDXAmount: Balance = 100_u128;
     pub const GetMaxSupply: Balance = 2_000_000_u128;
     pub const PolkadexIdoPalletId: PalletId = PalletId(*b"polk/ido");
+    pub const DefaultVotingPeriod : BlockNumber = 100_800;
```
Comment (simonsso) [inline runtime/src/lib.rs:1153]:
```suggestion
    pub const DefaultVotingPeriod : BlockNumber = 100_800; // One week
```

File: pallets/Polkadex_Ido/src/lib.rs [hunk 2]
```diff
@@ -521,6 +600,63 @@ decl_module! {
...
+            let who: T::AccountId = ensure_signed(origin)?;
+            ensure!(<InfoFundingRound<T>>::contains_key(&round_id), Error::<T>::FundingRoundDoesNotExist);
+            let funding_round = <InfoFundingRound<T>>::get(&round_id);
+            ensure!(current_block_no < funding_round.vote_end_block , Error::<T>::VotingEnded);
+            let voting = <RoundVotes<T>>::get(&round_id);
+            let position_yes = voting.ayes.iter().position(|a| a.account_id == who);
+            let position_no = voting.nays.iter().position(|a| a.account_id == who);
+            // Detects first vote of the member in the motion
+            let is_account_voting_first_time = position_yes.is_none() && position_no.is_none();
+            ensure!(is_account_voting_first_time, Error::<T>::DuplicateVote);
```
Comment (simonsso) [inline pallets/Polkadex_Ido/src/lib.rs:616]: A second ballot should remove the first vote and update with a new, or is this not how voting works elsewhere here?

File: pallets/Polkadex_Ido/src/lib.rs [hunk 3]
```diff
@@ -521,6 +600,63 @@ decl_module! {
...
+                    }
+                } else {
+                    if position_no.is_none() {
+                        voting.nays.push(voter);
+                    }
+                }
+            });
+
+            Ok(())
+        }
```
Comment (simonsso) [inline pallets/Polkadex_Ido/src/lib.rs:640]: Delegated votes. Will it work or do we need code for it?

File: pallets/Polkadex_Ido/src/test.rs [hunk 4]
```diff
@@ -609,3 +652,107 @@ fn test_withdraw_token() {
...
+        assert_eq!(
+            WhitelistInfoFundingRound::<Test>::contains_key(round_id),
+            false
+        );
+        assert_eq!(
+            InfoFundingRound::<Test>::contains_key(round_id),
+            false
+        );
+    });
+}
```
Comment (simonsso) [inline pallets/Polkadex_Ido/src/test.rs:758]: Would like to see test for vote where we saturate the internal counters and verify the full locked amount is returned.

File: pallets/Polkadex_Ido/src/lib.rs (hunk 2 above)
Comment (mambisi) [reply pallets/Polkadex_Ido/src/lib.rs:616]: Ok i think updating the ballot is a better design.

File: pallets/Polkadex_Ido/src/test.rs (hunk 4 above)
Comment (mambisi) [reply pallets/Polkadex_Ido/src/test.rs:758]: Ok i will add that

File: pallets/Polkadex_Ido/src/test.rs [hunk 5]
```diff
@@ -755,4 +755,34 @@ fn test_vote_for_round_no_vote_majority() {
...
+                balance,
+                open_block_number,
+                balance,
+                balance,
+                balance,
+                balance,
+                closing_block_number
+            ),
+            Ok(())
+        );
```
Comment (simonsso) [inline pallets/Polkadex_Ido/src/test.rs:789]: This test only tests if register_round returns ok. Name indicate it should test get reserve amount

Comment (mambisi) [reply pallets/Polkadex_Ido/src/test.rs:789]: https://github.com/Polkadex-Substrate/Polkadex/blob/a09408a6d98d9cfce5d5a9984eba4bfba3b6c49b/pallets/Polkadex_Ido/src/test.rs#L767

https://github.com/Polkadex-Substrate/Polkadex/blob/a09408a6d98d9cfce5d5a9984eba4bfba3b6c49b/pallets/Polkadex_Ido/src/test.rs#L796

https://github.com/Polkadex-Substrate/Polkadex/blob/a09408a6d98d9cfce5d5a9984eba4bfba3b6c49b/pallets/Polkadex_Ido/src/test.rs#L799

**Review by simonsso — APPROVED** [review]

**Pull request merged**